### DISCUSSION TITLE
Failure to start kernel due to failures related to pyzmq module

### DISCUSSION THREAD

**Discussion by yesdeepakmittal** [microsoft/vscode-jupyter#12545] [Questions and Answers]
I have reinstalled the pyzmq module several times as mentioned in this wiki - https://github.com/microsoft/vscode-jupyter/wiki/Failure-to-start-kernel-due-to-failures-related-to-pyzmq-module but I am still getting same 'Kernel failed to start due to pyzmq module' Error. Please suggest to me how to solve this issue.

**Comment by yesdeepakmittal** [comment]
I even tried by running JUPYTER NOTEBOOK as a docker container and connecting my VS CODE with it but still facing the same issue.

**Reply by DonJayamanne** [reply]
> I even tried by running JUPYTER NOTEBOOK as a docker container and connecting 

Can you run the kernel in Jupyter notebook web interface, without VS Code in this same docker container.

**Reply by yesdeepakmittal** [reply]
Since Jupyter notebook is running separately as a docker container, it is working fine there. But it is not working outside of Docker Environment.

**Reply by DonJayamanne** [reply]
> But it is not working outside of Docker Environment.

Can you try running the same Notebook with Jupyter Notebooks outside docker. 
This error is indicative of a faulty installation of a package. Hence the suggestion to try running Jupyter Notebooks outsidce the docker container, which would prove the fact that its not related to VS Code but a faulty install.

**Reply by yesdeepakmittal** [reply]
I tried several times running my Jupyter notebook outside of docker and in VSC. But this issue is not resolved. 

![image](https://user-images.githubusercontent.com/50262096/213865974-00a47761-da04-45cb-a118-1da8ca433398.png)
![image](https://user-images.githubusercontent.com/50262096/213867125-fc3ea5c5-b321-4b16-9083-216dd7164ab6.png)
